From 6484aebec707514360422273b688e443c6f0c3b2 Mon Sep 17 00:00:00 2001
From: Sage Weil <sage@redhat.com>
Date: Fri, 5 Dec 2014 14:21:08 -0800
Subject: [PATCH 11/22] common/ceph_context: don't import std namespace

This was broken by 7fed5dee4f96a83d1d6914f6fc0895bba2d15b99

Signed-off-by: Sage Weil <sage@redhat.com>
(cherry picked from commit 9029813b2784fe71879b1b7684855415b4c7566b)
(cherry picked from commit e391ebf3f471825c7efd84e61824b9e39d0e30c2)
---
 src/common/ceph_context.h | 5 ++---
 1 file changed, 2 insertions(+), 3 deletions(-)

diff --git a/src/common/ceph_context.h b/src/common/ceph_context.h
index e7b8b6c..7241c85 100644
--- a/src/common/ceph_context.h
+++ b/src/common/ceph_context.h
@@ -18,7 +18,6 @@
 #include <iostream>
 #include <stdint.h>
 #include <string>
-using namespace std;
 
 #include "include/buffer.h"
 #include "include/atomic.h"
@@ -109,7 +108,7 @@ public:
 		  bufferlist *out);
 
   template<typename T>
-  void lookup_or_create_singleton_object(T*& p, const string &name) {
+  void lookup_or_create_singleton_object(T*& p, const std::string &name) {
     ceph_spin_lock(&_associated_objs_lock);
     if (!_associated_objs.count(name)) {
       p = new T(this);
@@ -156,7 +155,7 @@ private:
   ceph::HeartbeatMap *_heartbeat_map;
 
   ceph_spinlock_t _associated_objs_lock;
-  map<string, AssociatedSingletonObject*> _associated_objs;
+  std::map<std::string, AssociatedSingletonObject*> _associated_objs;
 
   // crypto
   CryptoNone *_crypto_none;
-- 
2.1.0

